README.md

```diff
@@ -1,178 +1,194 @@
+#### Configuration Values ####
 
-| Parameter | Data Type | Description |
-| --- | --- | --- |
-| *error* | [Bayeux.Error](#bayeuxerror-class) | null if the connection is successful, error details otherwise. |
+These settings affect the client's behavior.
 
-#### Callback: onDisconnected(*reason*) ####
+| Key | Value Type | Required? | Description |
+| --- | --- | --- | --- |
+| *url* | String | Yes | The URL of the Bayeux server this client will connect to |
+| *backoffIncrement* | Integer | No | The number of seconds that the backoff time increments every time a connection with the Bayeux server fails. Bayeux.Client attempts to reconnect after the backoff time elapses. Default: 1 |
+| *maxBackoff* | Integer | No | The maximum number of seconds of the backoff time after which the backoff time is not incremented further. Default: 60 |
+| *requestTimeout* | Integer | No | The maximum number of seconds to wait before considering that a request to the Bayeux server failed. Default: 10 |
+| *requestHeaders* | Table | No | A key-value table containing the request headers to be sent for every Bayeux request, eg. `{ "My-Custom-Header" : "MyValue"}`. Default: `null` |
 
-This callback is called every time the client is disconnected.
+**Note** For more information on these settings, please see the [Bayeux protocol specification](https://docs.cometd.org/current/reference/#_bayeux).
 
-The client is considered as disconnected if any of these events occurs:
-- disconnection was caused by application
-- sending of "connect" message is failed (e.g., request timeout, HTTP error)
-- the last "connect" message is unsuccessful (response's "successful" field is "false") and has no "reconnect" advice set to "retry"
+#### Callback: onConnected ####
 
-To learn what "connect" message is, please read the [Bayeux protocol specification](https://docs.cometd.org/current/reference/#_bayeux).
+This callback is called every time the client connects. The client is considered to be connected when the handshake and the first *connect* messages were successful. To learn what handshake and *connect* messages are, please see the [Bayeux protocol specification](https://docs.cometd.org/current/reference/#_bayeux).
 
-This is a good place to call the [connect()](#connect) method again if it was an unexpected disconnection.
+The callback has one parameter of its own:
 
-| Parameter | Data Type | Description |
+| Parameter | Type | Description |
 | --- | --- | --- |
-| *reason* | [Bayeux.Error](#bayeuxerror-class) | null if the disconnection was caused by the [disconnect()](#disconnect) method, error details which explains a reason of the disconnection otherwise. |
+| *error* | [Bayeux.Error](#bayeuxerror-class) | This will be `null` if the connection is successful, otherwise an error message string |
```

Comment:
@smittytone We pass in to the callback an instance of Bayeux.Error class, not a string. We do so for all callbacks in this library.